lib/publiccloud/instances.pm

```diff
@@ -0,0 +1,26 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Base class for the public cloud namespace
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package publiccloud::instances;
+use testapi;
+use strict;
+use warnings;
+
+our @instances;    # Package variable containing all instanciated instances for global access without RunArgs
+
+sub set_instances {
+    @instances = @_;
+}
+
+sub get_instance {
+    return undef if (scalar @instances) < 1;
+    return $instances[0];
```

Comment:
this is a bit strange to me . We saying "ok we have collection of something" but getter method will give you only first item . Generally speaking I treat everything regarding "collection of instances" as dead code because we actually don't have cases where we create more than one entity . And I wish to drop this for a long time . But if you decided to define dedicated collection of instance getter suppose to give access to any element in collection

Reply:
I integrated this singleton variable in the current workflow, which allows for multiple instances. Don't blame this PR, it simply moves the existing variable handling into a global singleton that can be used also from other tests.

A refactoring of the underlying handling is IMHO way beyond the scope of this PR.

Reply:
ok

Reply:
Can we use this in the future to get rid of `$run_args`?

Reply:
AFAIK , it is not possible . TBH I am not sure how this code works . From what I know each module is executed separately so no matter what code structures you are creating in one module other one won't be able to reach them because after each module context is **destroyed**. This is why currently we passing everything through args of loadtest